enhancements/network/allow-mtu-changes.md

```diff
@@ -0,0 +1,291 @@
+possible time, which if under a reasonable time of 10 minutes, should be well
+under the typical TCP timeout interval.
+
+## Motivation
+
+While cluster administrators usually set the MTU correctly during the
+installation sometimes they need to change them afterwards for reasons such as
+changes in the underlay or because they were set incorrectly at install time.
+
+### Goals
+
+* Allow to change MTU post install on OVN Kubernetes.
+
+### Non goals
+
+* Change the MTU without service disruption.
+
+* Other safe or unsafe configuration changes.
+
+## Proposal
+
+The CNO monitors changes on the operator configuration. When it detects a MTU
+change:
+1. Set the `clusteroperator/network` conditions:
+   - Progressing: true
+   - Upgradeable: false
+2. Deploy pods on every node with `restartPolicy: Never` which are responsible
+   for validating the preconditions. If the preconditions are met the pod will
+   exit with code 0. Some of the preconditions that we will check are:
+   - For MTU changes, the new MTU is valid to apply on that node.
+3. Once all the previous pods finish successfully, deploy other set of pods with
+   `restartPolicy: Never` on every node that will handle the actual change of
+   the MTU (explained in more detail below). Wait for them to be ready and
+   running.
+4. Cordon every node, we don't want pods created from this point on.
+5. Ensure that the configmap ovnkube-config is synchronized with the new MTU
+   value.
+6. Set the new MTU value to the applied-cluster config map. At this point
+   CNO is commited to the MTU change.
+7. If any previous steps (1-6) was unsuccesful, the CNO will set the
+   `clusteroperator/network` conditions to:
+   - Progressing: false
+   - Degraded: true
+   Update the operator configuration status with a description of the problem.
+   At this point the process is interrupted and we require manual intervention.
+8. Wait for pods of step 3 to complete successfully.
+9. Force a rollout of the ovnkube-node daemonset. This will ensure
+   ovn-kubernetes uses the new MTU value for new pods as well as set the new MTU
+   on managed node interfaces like ovn-k8s-mp0, ovn-k8s-gw0 (local gateway mode)
+   and related routes.
+10. If any of the previous steps (8,9) failed, reboot the node, wait for the
+    kubelet to be reporting as Ready again and uncordon it.
+    If this step fails, set conditions to:
+    - Progressing: false
+    - Degraded: true
+    Update the operator configuration status with a description of the problem.
+    Otherwise just uncordon the node.
+11. Upon completion, set conditions to:
+    - Progressing: false
+    - Degraded: false
+
+The steps to change the MTU performed by pods of previous step 3 are:
+1. So that we don't have nodes doing things at different times and we have
+   everything synchronized, the pods will wait until the MTU value on the
+   applied-cluster configmap changes.
+2. Enter the network namespace of every pod in the pod network and change
```

Comment:
What do you mean by "in the pod network"? What I understand is that on every node we create a pod that enters every network namespace of that node and changes the MTU of eth0.